src/solvers/refinement/string_refinement.cpp

```diff
@@ -288,6 +289,25 @@ bool string_refinementt::add_axioms_for_string_assigns(
   return true;
 }
 
+/// Convert exprt to a specific type. Throw bad_cast if conversion
+/// cannot be performed
+/// Generic case doesn't exist, specialize for different types accordingly
+template<typename T>
```

Comment:
Yes, indeed, but I'm unsure whether that should go with this PR or with another (we've got refactor incoming, I have to stop adding changes).